guides/07_streaming/06_automatic-voice-detection.md

```diff
@@ -0,0 +1,244 @@
+# Multimodal Gradio App Powered by Groq with Automatic Speech Detection
+
+Tags: AUDIO, STREAMING, CHATBOTS, VOICE
+
+## Introduction
+Modern voice applications should feel natural and responsive, moving beyond the traditional "click-to-record" pattern. By combining Groq's fast inference capabilities with automatic speech detection, we can create a more intuitive interaction model where users can simply start talking whenever they want to engage with the AI.
+
+> Credits: VAD and Gradio code inspired by [WillHeld's Diva-audio-chat](https://huggingface.co/spaces/WillHeld/diva-audio-chat/tree/main).
+
+In this tutorial, you will learn how to create a multimodal Gradio and Groq app that has automatic speech detection. You can also watch the full video tutorial which includes a demo of the application:
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/azXaioGdm2Q" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
+
+## Background
+
+Many voice apps currently work by the user clicking record, speaking, then stopping the recording. While this can be a powerful demo, the most natural mode of interaction with voice requires the app to dynamically detect when the user is speaking, so they can talk back and forth without having to continually click a record button. 
+
+Creating a natural interaction with voice and text requires a dynamic and low-latency response. Thus, we need both automatic voice detection and fast inference. With @ricky0123/vad-web powering speech detection and Groq powering the LLM, both of these requirements are met. Groq provides a lightning fast response, and Gradio allows for easy creation of impressively functional apps.
+
+This tutorial shows you how to build a calorie tracking app where you speak to an AI that automatically detects when you start and stop your response, and provides its own text response back to guide you with questions that allow it to give a calorie estimate of your last meal.
+
+## Key Components
+
+- **Gradio**: Provides the web interface and audio handling capabilities
+- **@ricky0123/vad-web**: Handles voice activity detection
+- **Groq**: Powers fast LLM inference for natural conversations
+- **Whisper**: Transcribes speech to text
+
+### Setting Up the Environment
+
+First, let’s import our essential libraries and set up a client for using the Groq API. Here’s how to do it:
+
+```python
+import groq
```

Comment:
Would be nice to include the contents of a `requirements.txt` file for users to get started quickly